src/radiation_system.hpp

```diff
@@ -630,11 +637,21 @@ AMREX_GPU_DEVICE auto RadSystem<problem_t>::ComputeCellOpticalDepth(const quokka
 	} else if constexpr (DIR == FluxDir::X3) {
 		dl = dx[2];
 	}
-	quokka::valarray<double, nGroups_> const tau_L = dl * rho_L * RadSystem<problem_t>::ComputeFluxMeanOpacity(rho_L, Tgas_L);
-	quokka::valarray<double, nGroups_> const tau_R = dl * rho_R * RadSystem<problem_t>::ComputeFluxMeanOpacity(rho_R, Tgas_R);
 
-	return (tau_L * tau_R * 2.) / (tau_L + tau_R); // harmonic mean
-						       // return 0.5*(tau_L + tau_R); // arithmetic mean
+	quokka::valarray<double, nGroups_> optical_depths{};
+	if constexpr (nGroups_ == 1) {
```

Comment:
That's not the case. It will work for both single-group and multigroup. That's what the if constexpr (nGroups_ == 1) is doing. Note that DefineOpacityExponentsAndLowerValues and ComputeBinCenterOpacity returns an valarray, so optical_depths is a valarray of length nGroups_.

Reply:
Although, I'm kind of cutting corner here by using ComputeBinCenterOpacity regardless of the OpacityModel the user chooses to use. But I think this won't matter since this wavespeed correction is only an approximation method.